src/main/scala/is/hail/expr/ir/functions/ReferenceGenomeFunctions.scala

```diff
@@ -99,47 +99,62 @@ class ReferenceGenomeFunctions(rg: ReferenceGenome) extends RegistryFunctions {
     registerCode(newName, args, rt, isDeterministic)(impl)
   }
 
-  def registerRGCode(
+  def registerRGCode[A1](
```

Comment:
nice, nice.

Scala, as you've discovered, really struggles with existentials. I think we basically should always use universals in favor of existentials, as you've done here.